core/src/main/java/org/commonjava/indy/core/ctl/ContentController.java

```diff
@@ -530,5 +530,17 @@ public void generateHttpMetadataHeaders( Transfer target, final HttpServletReque
             IOUtils.closeQuietly( out );
         }
 
+        // npm will generate .tgz and version json metadata files from the package json file target,
+        // which will also need the HttpExchangeMetadata for npm header check.
+
+        Set<Transfer> generated = target.getGeneratedFromTarget();
```

Comment:
Looking more closely at this, I think this behavior should be handled inside Indy's NPM support modules. I mentioned this in the galley PR too.